keanu-project/src/main/java/io/improbable/keanu/util/DescriptionCreator.java

```diff
@@ -0,0 +1,178 @@
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.DifferenceVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.DivisionVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.MultiplicationVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.IntegerIfVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.operators.binary.IntegerAdditionVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.operators.binary.IntegerDifferenceVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.operators.binary.IntegerMultiplicationVertex;
+import io.improbable.keanu.vertices.intgr.probabilistic.BinomialVertex;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static io.improbable.keanu.util.DescriptionUtils.createBinomialDescription;
+import static io.improbable.keanu.util.DescriptionUtils.createBooleanBinaryOpDescription;
+import static io.improbable.keanu.util.DescriptionUtils.createIfStringDescription;
+
+public class DescriptionCreator {
+
+    private static Map<Class, String> delimiters = new HashMap<>();
+    private static Map<Class, String> booleanBinaryOperators = new HashMap<>();
+
+    static {
+        delimiters.put(AdditionVertex.class, " + ");
+        delimiters.put(IntegerAdditionVertex.class, " + ");
+        delimiters.put(DifferenceVertex.class, " - ");
+        delimiters.put(IntegerDifferenceVertex.class, " - ");
+        delimiters.put(MultiplicationVertex.class, " * ");
+        delimiters.put(IntegerMultiplicationVertex.class, " * ");
+        delimiters.put(DivisionVertex.class, " / ");
+        booleanBinaryOperators.put(AndBinaryVertex.class, " && ");
+        booleanBinaryOperators.put(EqualsVertex.class, " == ");
+        booleanBinaryOperators.put(GreaterThanOrEqualVertex.class, " >= ");
+        booleanBinaryOperators.put(GreaterThanVertex.class, " > ");
+        booleanBinaryOperators.put(LessThanOrEqualVertex.class, " <= ");
+        booleanBinaryOperators.put(LessThanVertex.class, " < ");
+        booleanBinaryOperators.put(NotEqualsVertex.class, " != ");
+        booleanBinaryOperators.put(OrBinaryVertex.class, " || ");
+    }
+
+    private static <T extends Tensor> String getBaseDescription(Vertex<T> vertex) {
+        if (vertex instanceof ConstantVertex) {
+            Tensor value = vertex.getValue();
+            String valueString = value.isScalar() ? value.scalar().toString() :
+                new StringBuilder("(Tensor with shape: ").append(Arrays.toString(value.getShape())).append(")").toString();
+
+            return new StringBuilder()
+                .append("Const(")
+                .append(valueString)
+                .append(")")
+                .toString();
+        } else {
+            return "<<Unlabelled leaf Vertex>>";
```

Comment:
In general I'd shy away from logic like this.  In particular, this function makes assumptions about how other functions behave.  Whilst correct at the moment, someone might not recognise there's an implicit dependency between those two functions.

Reply:
@christophernorth I'm not sure I understand - which function is this logic making assumptions about?